### PR TITLE
add aws credential provider docs

### PR DESCRIPTION
**Pull request by freeznet**
Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

Add docs describe options for `awsCredentialPluginName`.

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [x] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: docs/sqs-sink.md [hunk 1]
```diff
@@ -46,13 +46,15 @@ Before using the SQS sink connector, you need to configure it. Below are the pro
...
+|----------------------------|--------|----------|--------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `awsEndpoint`              | String | false    | " " (empty string) | AWS SQS end-point URL. It can be found at [here](https://docs.aws.amazon.com/general/latest/gr/sqs-service.html#sqs_region).                                                                                                                                                                                                                                 |
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:51]:
```suggestion
| `awsEndpoint`              | String | false    | " " (empty string) | AWS SQS end-point URL. You can find it at [AWS SQS Service endpoints](https://docs.aws.amazon.com/general/latest/gr/sqs-service.html#sqs_region).                                                                                                                                                                                                                                 |
```

File: docs/sqs-sink.md [hunk 2]
```diff
@@ -46,13 +46,15 @@ Before using the SQS sink connector, you need to configure it. Below are the pro
...
+| `awsRegion`                | String | true     | " " (empty string) | Supported AWS region. For example, us-west-1, us-west-2.                                                                                                                                                                                                                                                                                                     |
+| `awsCredentialPluginName`  | String | false    | " " (empty string) | Fully-qualified class name of implementation of `AwsCredentialProviderPlugin`, builtin options are listed below. It is a factory class which creates an AWSCredentialsProvider that is used by SQS connector. If it is empty, the SQS connector creates a default AWSCredentialsProvider which accepts json-map of credentials in `awsCredentialPluginParam` |
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:53]:
```suggestion
| `awsCredentialPluginName`  | String | false    | " " (empty string) | Fully-qualified class name of implementation of `AwsCredentialProviderPlugin`. Built-in options are listed below. It is a factory class that creates an AWSCredentialsProvider that is used by the SQS connector. If it is empty, the SQS connector creates a default AWSCredentialsProvider which accepts a JSON-format map of credentials in `awsCredentialPluginParam`. |
```

File: docs/sqs-sink.md [hunk 3]
```diff
@@ -46,13 +46,15 @@ Before using the SQS sink connector, you need to configure it. Below are the pro
...
+| `awsCredentialPluginParam` | String | true     | " " (empty string) | JSON parameter to initialize `AwsCredentialsProviderPlugin`.                                                                                                                                                                                                                                                                                                 |
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:54]:
```suggestion
| `awsCredentialPluginParam` | String | true     | " " (empty string) | The JSON parameter to initialize `AwsCredentialsProviderPlugin`.                                                                                                                                                                                                                                                                                                 |
```

File: docs/sqs-sink.md [hunk 4]
```diff
@@ -46,13 +46,15 @@ Before using the SQS sink connector, you need to configure it. Below are the pro
...
+| `queueName`                | String | true     | " " (empty string) | Name of the SQS queue that messages should be read from or written to.                                                                                                                                                                                                                                                                                       |
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:55]:
```suggestion
| `queueName`                | String | true     | " " (empty string) | The name of the SQS queue that messages should be read from or written to.                                                                                                                                                                                                                                                                                       |
```

File: docs/sqs-sink.md [hunk 5]
```diff
@@ -61,6 +63,39 @@ following permissions to Amazon SQS API:
 - sqs:CreateQueue
 - sqs:SendMessage
 
+### Built-in AWS Credential plugins
+
+The following are built-in `AwsCredentialProviderPlugin` plugins:
+
+* ` ` (empty)
+
+  If the plugin is empty, the SQS connector creates a default AWSCredentialsProvider which accepts json-map of credentials in `awsCredentialPluginParam`.
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:72]:
```suggestion
  If the plugin is empty, the SQS connector creates a default AWSCredentialsProvider which accepts a JSON-format map of credentials in `awsCredentialPluginParam`.
```

File: docs/sqs-sink.md [hunk 6]
```diff
@@ -61,6 +63,39 @@ following permissions to Amazon SQS API:
...
+
+  This plugin takes no configuration, it uses the default AWS provider chain.
+
+  For more information, see [AWS documentation](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default).
+
+* `org.apache.pulsar.io.aws.STSAssumeRoleProviderPlugin`
+
+  This plugin takes a configuration (via the `awsCredentialPluginParam`) that describes a role to assume when running the SQS Client.
+
+  This configuration takes the form of a small json document like:
```
Comment (Huanli-Meng) [inline docs/sqs-sink.md:93]:
```suggestion
  This configuration takes the form of a small JSON-format document like below:
```

**Review by Huanli-Meng — APPROVED** [review]
Leave some comments on the sink doc. Please also update the comment to the source doc. Thanks.

**Review by nlu90 — APPROVED** [review]

**Pull request merged**